### ISSUE TITLE
[Platform] Improve performance of ENS name lookup

### ISSUE THREAD

**Issue by vikmeup** [trustwallet/blockatlas#579]
Currently it's quite slow to get ENS name, we need to find better ways to resolve it

**Comment by arein** [comment]
Do you have some implementation proposal?

**Comment by Pantani** [comment]
@vikmeup the issue is inside the ENS package (`github.com/wealdtech/go-ens/v3`), I did this benchmark about the response time of these 2 methods

[resolver, err := ens.NewResolver(client, name)](https://github.com/trustwallet/blockatlas/blob/03cb7ab08475d62ad4aca1a9838594f7dc02f4d8/platform/ethereum/domain.go#L20)

[address, err := resolver.MultiAddress(coin)](https://github.com/trustwallet/blockatlas/blob/03cb7ab08475d62ad4aca1a9838594f7dc02f4d8/platform/ethereum/domain.go#L38)

The result:
```
INFO[0009] ens.NewResolver method                        fields.time=3.396843919
INFO[0012] resolver.MultiAddress method                  fields.time=3.027379971
[GIN] 2020/02/19 - 21:15:19 | 200 |  7.581521737s |             ::1 | GET      /ns/lookup?name=vitalik.luxe&coin=60

INFO[0018] ens.NewResolver method                        fields.time=1.643374428
INFO[0020] resolver.MultiAddress method                  fields.time=2.307362197
[GIN] 2020/02/19 - 21:15:26 | 200 |  4.102615409s |             ::1 | GET      /ns/lookup?name=vitalik.luxe&coin=60

INFO[0034] ens.NewResolver method                        fields.time=10.67012079
INFO[0034] resolver.MultiAddress method                  fields.time=0.323845119
[GIN] 2020/02/19 - 21:15:40 | 200 | 11.146582641s |             ::1 | GET      /ns/lookup?name=vitalik.luxe&coin=60

INFO[0063] ens.NewResolver method                        fields.time=1.6460560499999999
INFO[0064] resolver.MultiAddress method                  fields.time=1.314818209
[GIN] 2020/02/19 - 21:16:10 | 200 |  3.117270452s |             ::1 | GET      /ns/lookup?name=vitalik.luxe&coin=60

INFO[0067] ens.NewResolver method                        fields.time=1.660254068
INFO[0067] resolver.MultiAddress method                  fields.time=0.318249766
[GIN] 2020/02/19 - 21:16:13 | 200 |  2.131854541s |             ::1 | GET      /ns/lookup?name=vitalik.luxe&coin=60
```

**Comment by Pantani** [comment]
I think is the same issue

https://github.com/trustwallet/blockatlas/issues/512

**Comment by EnoRage** [comment]
solved by #973
